### PR TITLE
GH-38946: [MATLAB] Fix spelling

### LINKED ISSUE #38946: [MATLAB] Spelling errors identified by check-spelling
### Describe the bug, including details regarding any error messages, version, and platform.

The [check-spelling action](https://github.com/marketplace/actions/check-spelling) enabled me to identify some misspellings...

The misspellings have been reported at https://github.com/jsoref/apache-arrow/actions/runs/7008141484/attempts/1#summary-19063787889

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by jsoref**
### What changes are included in this PR?

Spelling fixes to matlab/

### Are there any user-facing changes?

* Closes: #38946

### CODE REVIEW

File: matlab/test/arrow/array/tTime64Array.m [hunk 1]
```diff
@@ -128,10 +128,10 @@ function TestDuration(testCase, Unit)
         function TestValid(testCase, Unit)
             % Verify the Valid property returns the expected logical vector.
             times = seconds([100 200 NaN 355 NaN 400]);
-            arrray = testCase.ArrowArrayConstructorFcn(times, TImeUnit=Unit);
-            testCase.verifyEqual(arrray.Valid, [true; true; false; true; false; true]);
-            testCase.verifyEqual(toMATLAB(arrray), times');
-            testCase.verifyEqual(duration(arrray), times');
+            array = testCase.ArrowArrayConstructorFcn(times, TImeUnit=Unit);
```
Comment (kou) [inline matlab/test/arrow/array/tTime64Array.m:131]: "TImeUnit" may be a typo too.

```diff
-TImeUnit
+TimeUnit
```

**Review by kou — APPROVED** [review]
+1

File: matlab/test/arrow/array/tTime64Array.m (hunk 1 above)
Comment (jsoref) [reply matlab/test/arrow/array/tTime64Array.m:131]: Oh, probably. 

In some places I think I detected/added a rule to tolerate `TFoo`, and Ime is sometimes a form of IME (input method editor), but, you're right, that doesn't seem likely to apply here.

File: matlab/test/arrow/array/tChunkedArray.m [hunk 2]
```diff
@@ -112,7 +112,7 @@ function TestIsEqualTrue(testCase)
             % Verifies ChunkedArrays are considered equal if:
             %
             %  1. Their Type properties are equal
-            %  2. Their NumElements properties ar eequal
+            %  2. Their NumElements properties ar equal
```
Comment (kevingurney) [inline matlab/test/arrow/array/tChunkedArray.m:115]: ar equal -> are equal

Comment (kevingurney) [inline matlab/test/arrow/array/tChunkedArray.m:115]:
```suggestion
            %  2. Their NumElements properties are equal
```

**Review by kevingurney — CHANGES_REQUESTED** [review]

**Review by kevingurney — APPROVED** [review]

**Comment by kevingurney** [comment]
+1

**Pull request merged**